ddev/src/ddev/cli/validate/version.py

```diff
@@ -0,0 +1,144 @@
+import re
+from functools import partial
+from typing import TYPE_CHECKING
+
+import click
+
+if TYPE_CHECKING:
+    from ddev.cli.application import Application
+
+MISSING = object()
+
+
+def _parse_changelog(path):
+    raw = path.read_text()
+    managed_by_towncrier = 'towncrier release notes start' in raw
+    releases = re.findall(r'## (\d+\.\d+\.\d+)', raw)
+    latest = releases[0] if releases else MISSING
+    return managed_by_towncrier, latest
+
+
+def _validate_python_package(
+    track_err, repo_path, current_check, version_file, changelog, changelog_is_autogenerated, changelog_latest_v
+):
+    if not changelog_is_autogenerated:
+        track_err(
+            message=(
+                f'This looks like a Python package, but {changelog.relative_to(repo_path)} is managed manually. '
+                'Please add the towncrier header to the CHANGELOG.'
+            )
+        )
+        return
+
+    latest_pkg_v = re.findall(r"(\d+\.\d+\.\d+)", version_file.read_text())[0]
```

Comment:
Oh, wait, you mean that the file exists but is just an empty file? I considered that but thought it's highly unlikely: we only add that file as part of a template in `ddev create` and it's never empty there even if it may have the wrong version.

Good question, I was really tempted to add a test for it, but thought mb that would be too much edge-case testing 🤓

Reply:
Yeah I meant file exists but is empty, I figured you could do something like [this](https://github.com/DataDog/integrations-core/pull/18063/files/64683f510fa249ca51953edca1848106ca197a1c#diff-6fea52f19b2e0a5a2851c05a68367a577122adac55b2b6e0d7e306794c8799dbR16-R17) and error out if it is empty.

Reply:
I think I'll leave it as is for now and see what feedback we get. Don't wanna overengineer it too much.

Reply:
Good work on spotting this, keep that eye sharp. Just cuz we're punting on it this time, doesn't mean we always will ;)